assets/konnectivity/konnectivity-agent-data-plane.yaml

```diff
@@ -0,0 +1,86 @@
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: konnectivity-agent
+  namespace: kube-system
+  labels:
+    app: konnectivity-agent
+    kubernetes.io/cluster-service: "true"
+spec:
+  selector:
+    matchLabels:
+      app: konnectivity-agent
+  updateStrategy:
+    rollingUpdate:
+      maxUnavailable: 1
+      maxSurge: 0
+  template:
+    metadata:
+      labels:
+        app: konnectivity-agent
+    spec:
+      automountServiceAccountToken: false
+      priorityClassName: system-node-critical
+      containers:
+      - name: konnectivity-agent
+        image: {{ imageFor "apiserver-network-proxy" }}
+        imagePullPolicy: IfNotPresent
+{{- if .KonnectivitySecurityContext }}
+{{- $securityContext := .KonnectivitySecurityContext }}
+        securityContext:
+          runAsUser: {{ $securityContext.RunAsUser }}
+{{- end }}
+        command: ["/bin/proxy-agent"]
+        args: [
+          "--logtostderr=true",
+          "--ca-cert=/etc/konnectivity/ca/ca.crt",
+          "--proxy-server-host={{ .KonnectivityServerURL }}",
+          "--proxy-server-port={{ .KonnectivityServerAgentNodePort }}",
+          "--health-server-port={{ .KonnectivityAgentHealthPort }}",
+          "--agent-cert=/etc/konnectivity/agent/tls.crt",
+          "--agent-key=/etc/konnectivity/agent/tls.key",
+          "--agent-identifiers=ipv4=$(HOST_IP)&default-route=true",
```

Comment:
Both are correct.
We have this strategy in server: "--proxy-strategies=destHost,defaultRoute",

This means when the apiserver tries to get logs, it can go directly to the right node (as we destHost strategy). If it cannot, it can use any of the agents (defaultRoute). To have feature parity with Hypershift, I remove this for now, but it could be a good enhance. Thanks!